src/components/ProtectButton/index.tsx

```diff
@@ -0,0 +1,158 @@
+/**
+ * Copyright (c) Flashbots Ltd. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import {PropsWithChildren} from 'react';
+import type {HintPreferences} from '@flashbots/mev-share-client';
+import {useSDK} from '@metamask/sdk-react';
+import styles from './styles.module.scss';
+
+const RPC_GOERLI_FLASHBOTS_NET = 'https://rpc-goerli.flashbots.net';
+const RPC_SEPOLIA_FLASHBOTS_NET = 'https://rpc-sepolia.flashbots.net';
+const RPC_FLASHBOTS_NET = 'https://rpc.flashbots.net';
+
+export const mungeHintsForRpcUrl = (hints: HintPreferences) => {
+  /*
+    `hash` is always shared on the backend.
+    We only need to specify it if we don't want default hints shared.
+
+    If other hints are specified, `hash` is implied. In that case we
+    set hash to undefined so it's removed from the URL.
+ */
+  const hashImplied = Object.values(hints).some((v) => v);
+  return {
+    calldata: hints.calldata,
+    contract_address: hints.contractAddress,
+    function_selector: hints.functionSelector,
+    logs: hints.logs,
+    default_logs: hints.defaultLogs,
+    hash: hashImplied ? false : hints.txHash,
+  };
+};
+
+export interface ProtectButtonOptions extends PropsWithChildren {
+  /** Specify data to share; if undefined, uses default [Stable config](https://docs.flashbots.net/flashbots-protect/rpc/mev-share#stable-configuration) */
+  hints: HintPreferences;
+  /** Selected builders that are permitted to build blocks using the client's transactions. */
+  builders: Array<string>;
+  /** `fast` mode enables all supported builders implicitly. Setting `fast` will override `builders`. */
+  fast: boolean;
+}
+
+export const generateRpcUrl = ({
+  chainId,
+  options: {hints, builders, fast},
+}: {
+  chainId: string;
```

Comment:
chainId as a string is kinda unusual -- why not have it be a number?

Reply:
oh it is the format provided by MetaMask SDK. So I stuck with that.